apps/taxes/src/pages/api/webhooks/order-created.ts

```diff
@@ -0,0 +1,109 @@
+import { SaleorAsyncWebhook } from "@saleor/app-sdk/handlers/next";
+import {
+  OrderCreatedEventSubscriptionFragment,
+  OrderStatus,
+  UntypedOrderCreatedSubscriptionDocument,
+  UpdateMetadataDocument,
+  UpdateMetadataMutation,
+  UpdateMetadataMutationVariables,
+} from "../../../../generated/graphql";
+import { saleorApp } from "../../../../saleor-app";
+import { createLogger } from "../../../lib/logger";
+import { getActiveTaxProvider } from "../../../modules/taxes/active-tax-provider";
+import { createClient } from "../../../lib/graphql";
+import { Client } from "urql";
+import { WebhookResponse } from "../../../modules/app/webhook-response";
+
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+};
+
+type OrderCreatedPayload = Extract<
+  OrderCreatedEventSubscriptionFragment,
+  { __typename: "OrderCreated" }
+>;
+
+export const orderCreatedAsyncWebhook = new SaleorAsyncWebhook<OrderCreatedPayload>({
+  name: "OrderCreated",
+  apl: saleorApp.apl,
+  event: "ORDER_CREATED",
+  query: UntypedOrderCreatedSubscriptionDocument,
+  webhookPath: "/api/webhooks/order-created",
+});
+
+// * This is the key that we use to store the provider order id in the Saleor order metadata.
+export const PROVIDER_ORDER_ID_KEY = "externalId";
+
+/**
+ * We need to store the provider order id in the Saleor order metadata so that we can
+ * update the provider order when the Saleor order is fulfilled.
+ */
+async function updateOrderMetadataWithExternalId(
+  client: Client,
+  orderId: string,
+  externalId: string
+) {
+  const { error } = await client
+    .mutation<UpdateMetadataMutation>(UpdateMetadataDocument, {
+      id: orderId,
+      input: [
+        {
+          key: PROVIDER_ORDER_ID_KEY,
+          value: externalId,
+        },
+      ],
+    } as UpdateMetadataMutationVariables)
```

Comment:
```
{
      id: orderId,
      input: [
        {
          key: PROVIDER_ORDER_ID_KEY,
          value: externalId,
        },
      ],
    } as UpdateMetadataMutationVariables
```

in typescript its pretty dangerous operation because you first write value, then say to typescript "Im sure its ok, don't check it"

instead, it should be 

```
const variables: UpdateMetadataMutationVariables = {
      id: orderId,
      input: [
        {
          key: PROVIDER_ORDER_ID_KEY,
          value: externalId,
        },
      ],
} 

client.mutation(UpdateMetadataDocument, variables) 

```